### PR TITLE
script to push completed submissions in survey submission collection

### PR DESCRIPTION
**Pull request by borkarsaish65**

### CODE REVIEW

File: scripts/survey/pushcompletedsubmissions.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        .collection("surveySubmissions")
+        .find({solutionId:{$in:IDArray.map(id=>ObjectId(id))},status:"completed"})
+        .toArray(); 
+
+      let secondRecordSet = await db
+        .collection("surveySubmissions")
+        .find({_id:{$in:IDArray.map(id=>ObjectId(id))},status:"completed"})
+        .toArray(); 
+
+    if(firstRecordSet && firstRecordSet.length > 0)
```
Comment (VISHNUDAS-tunerlabs) [inline scripts/survey/pushcompletedsubmissions.js:46]: @borkarsaish65 If they pass both solutionId and submissionId conditions won't be executed. Please check how they will pass the ids. Based on discussion if changes required please add it here.

File: scripts/survey/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+This script is intended as a remedy to push completed surveySubmission records to kafka and it is related to the 6.0.0 branch
+
+### Step 1:
+
+    Navigate to /opt/projects/scripts/survey/
+
+### Step 2:
+
+Run the script to delete duplicate projects.
```
Comment (VISHNUDAS-tunerlabs) [inline scripts/survey/README.md]: @borkarsaish65 this is wrong. please update

File: scripts/survey/README.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
+## Migrations
+
+#### Steps to run the script files
+
+This script is intended as a remedy to push completed surveySubmission records to kafka and it is related to the 6.0.0 branch
+
+### Step 1:
+
+    Navigate to /opt/projects/scripts/survey/
```
Comment (VISHNUDAS-tunerlabs) [inline scripts/survey/README.md]: @borkarsaish65 generalise this path.

File: scripts/survey/README.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    note: id here can be submission_id that is _id from surveySubmission collection or solution_id based on which it will query the surveySubmission collection to find the records.
+
+    Multiple IDs support:
+
+    You can also provide multiple IDs by separating them with commas.
+    Example: node pushcompletedsubmissions.js id1,id2,id3
+
+#### Validation 
+
+After the script has been executed in the staging environment, all the pushed topics to kafka those ids will be stored in "pushedTopics.txt" file.
```
Comment (VISHNUDAS-tunerlabs) [inline scripts/survey/README.md]: @borkarsaish65 please correct the description as we discussed

File: scripts/survey/pushcompletedsubmissions.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+          if (pushAction.status == "success") {
+            successArray.push(currentBatch[j]._id);
+          }
+        } catch (err) {
+          console.log(err, "Error caught");
+        }
+      }
+    }
+    
+    fs.writeFile('./scripts/survey/pushedTopics'+utils.generateUUId()+'.txt', JSON.stringify(successArray), (err) => {
```
Comment (VISHNUDAS-tunerlabs) [inline scripts/survey/pushcompletedsubmissions.js]: @borkarsaish65 please change the file name to pushedMessageIds. We will be pushing all these data to a single topic

**Review by VISHNUDAS-tunerlabs — CHANGES_REQUESTED** [review]
Reviewed at 10:30 AM : Jul 22 - 2024

File: scripts/survey/pushcompletedsubmissions.js (hunk 1 above)
Comment (borkarsaish65) [reply scripts/survey/pushcompletedsubmissions.js:46]: Checked with Prashant , only submission_ids will be pushed.

File: scripts/survey/pushcompletedsubmissions.js (hunk 5 above)
Comment (borkarsaish65) [reply scripts/survey/pushcompletedsubmissions.js]: done

File: scripts/survey/README.md (hunk 2 above)
Comment (borkarsaish65) [reply scripts/survey/README.md]: changed

File: scripts/survey/README.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
+## Migrations
+
+#### Steps to run the script files
+
+This script is intended as a remedy to push completed surveySubmission records to kafka and it is related to the 6.0.0 branch
+
+### Step 1:
+
+    Navigate to Project Directory
```
Comment (VISHNUDAS-tunerlabs) [inline scripts/survey/README.md:9]: @borkarsaish65 should we add from project directory to which directory to navigate?

**Review by VISHNUDAS-tunerlabs — CHANGES_REQUESTED** [review]

**Review by VISHNUDAS-tunerlabs — APPROVED** [review]
Reviewed [jul 22 - 11:05]

**Pull request merged**